### PR TITLE
Fix prettier 1.5.x

### PR DESCRIPTION
**Pull request by akupila**
Prettier formats the code so that

```jsx
import styles from './styles';

export default () =>
  <div>
    <h1 className="test">Hello</h1>
    <style jsx>{styles}</style>
  </div>;
```

becomes

```jsx
import styles from './styles';

export default () =>
  <div>
    <h1 className="test">Hello</h1>
    <style jsx>
      {styles}
    </style>
  </div>;
```

This led to a confusing error `Invalid attempt to destructure non-iterable instance`:

image

@giuseppeg [proposed a fix for this problem](https://github.com/zeit/styled-jsx/issues/251#issuecomment-311453600), this PR is just exactly that fix + a test that fails before the fix.

In addition there is another commit which bumps prettier to 1.5.2 and formats that test fixture using prettier (so that future updates to prettier should still work). Not sure if this PR should be the place for it, if not i can remove the commit

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "prepublish": "gulp transpile",
     "test": "npm run lint && ava",
     "lint": "xo",
-    "format": "prettier --single-quote --no-semi --write *.js {src,test}/*.js",
+    "format": "prettier --single-quote --no-semi --write *.js {src,test}/*.js test/fixtures/*-prettier.js",
```
Comment (giuseppeg) [inline package.json]: i don't think that we need to format the fixture, the bug can be reproduced without prettier. can you revert this change?

File: test/index.js [hunk 2]
```diff
@@ -93,6 +93,11 @@ test('works with external stylesheets', async t => {
   t.snapshot(code)
 })
 
+test('works with external stylesheets (prettier)', async t => {
+  const { code } = await transform('./fixtures/external-stylesheet-prettier.js')
+  t.snapshot(code)
```
Comment (giuseppeg) [inline test/index.js]: I guess that you can merge this with the existing test for external styles. If you want to keep them separate then maybe we should rename it because as I said early the bug it is not about prettier

File: test/fixtures/external-stylesheet-prettier.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import styles from "./styles"
+
+export default () =>
+  <div>
+    <p>test</p>
+    <style jsx>
+      {styles}
+    </style>
```
Comment (giuseppeg) [inline test/fixtures/external-stylesheet-prettier.js]: can you add a few test cases more?

```jsx
<style jsx>{

styles}
</style>
<style jsx>{

styles
}
</style>
<style jsx>

{

styles}
</style>
<style jsx>{styles}
</style>
<style jsx>

{styles}</style>
<style jsx>


{styles}

</style>

<style jsx>{'div { color: red}'}

{styles}
</style>
```

**Comment by leo** [comment]
Thank you very much, @akupila! I'll make sure to mention you in the GitHub release.

However, we needed this fixed very quickly, so I had to introduce the changes by myself in https://github.com/zeit/styled-jsx/pull/256. I hope you can understand that! 😊

File: package.json (hunk 1 above)
Comment (akupila) [inline package.json]: The thinking was that if a prettier 1.6 comes out we want to ensure it doesn't break the babel plugin parsing. A bit of an edge case i guess

File: test/fixtures/external-stylesheet-prettier.js (hunk 3 above)
Comment (akupila) [inline test/fixtures/external-stylesheet-prettier.js]: Oh, yeah. Should have done that but was in a hurry :)

**Comment by akupila** [comment]
Damn, missed the notification about responses and was too slow to respond :) That's great though, good to see this merged 🎉

**Comment by akupila** [comment]
I think the additional tests @giuseppeg pointed out would have been good to add though before merging

**Comment by giuseppeg** [comment]
thank you guys for taking this, i am on holiday and without a laptop

**Comment by leo** [comment]
@akupila I don't have time right now and we need the fix to be released (will do in a sec). But feel free to do another PR, if you want 😊

File: package.json (hunk 1 above)
Comment (giuseppeg) [inline package.json]: the thing is that @leo found this bug because of prettier but the formatting which caused the issue can also be done by hand

Comment (akupila) [inline package.json]: True. I think formatting a fixture as part of the the `format` is also something i don't really like.. -> Totally ok to have it removed :)
